src/app/Console/Commands/Install.php

```diff
@@ -26,45 +31,191 @@ class Install extends Command
 
+    /**
+     * Addons variable.
+     *
+     * @var array
+     */
+    protected $addons = [
+        Addons\RequirePro::class,
+        Addons\RequireDevTools::class,
+        Addons\RequireEditableColumns::class,
+    ];
+
     /**
      * Execute the console command.
      *
      * @return mixed Command-line output
      */
     public function handle()
     {
-        $this->progressBar = $this->output->createProgressBar(5);
-        $this->progressBar->minSecondsBetweenRedraws(0);
-        $this->progressBar->maxSecondsBetweenRedraws(120);
-        $this->progressBar->setRedrawFrequency(1);
-
-        $this->progressBar->start();
-
-        $this->info(' Backpack installation started. Please wait...');
-        $this->progressBar->advance();
+        $this->infoBlock('Installing Backpack CRUD:', 'Step 1');
 
-        $this->line(' Publishing configs, views, js and css files');
+        // Publish files
+        $this->progressBlock('Publishing configs, views, js and css files');
         $this->executeArtisanProcess('vendor:publish', [
-            '--provider' => 'Backpack\CRUD\BackpackServiceProvider',
+            '--provider' => BackpackServiceProvider::class,
             '--tag' => 'minimum',
         ]);
+        $this->closeProgressBlock();
 
-        $this->line(" Creating users table (using Laravel's default migration)");
+        // Create users table
+        $this->progressBlock('Creating users table');
         $this->executeArtisanProcess('migrate', $this->option('no-interaction') ? ['--no-interaction' => true] : []);
+        $this->closeProgressBlock();
 
-        $this->line(" Creating App\Http\Middleware\CheckIfAdmin.php");
+        // Create CheckIfAdmin middleware
+        $this->progressBlock('Creating CheckIfAdmin middleware');
         $this->executeArtisanProcess('backpack:publish-middleware');
+        $this->closeProgressBlock();
+
+        // Install Backpack Generators
+        $this->progressBlock('Installing Backpack Generators');
+        $process = new Process(['composer', 'require', '--dev', 'backpack/generators']);
+        $process->setTimeout(300);
+        $process->run();
+        $this->closeProgressBlock();
+
+        // Create users
+        $this->createUsers();
+
+        // Addons
+        $this->installAddons();
+
+        // Done
+        $url = Str::of(config('app.url'))->finish('/')->append('admin/');
+        $this->infoBlock('Backpack installation complete.', 'done');
+        $this->note("Go to <fg=blue>$url</> to access your new admin panel.");
+        $this->note('You may need to run `php artisan serve` to serve your Laravel project.');
+        $this->newLine();
+    }
+
+    private function createUsers()
+    {
+        $userClass = config('backpack.base.user_model_fqn', 'App\Models\User');
+        $userModel = new $userClass();
+
+        // Count current users
+        $currentUsers = $userModel->count();
```

Comment:
Please let's take into consideration that users could contain millions of records and this `count()` would be a very expensive operation in that case.

I was about to suggest to use an "optimized query count" here, but thinking better about it, **we should remove it completely**. 😿 
I know it's a very nice feedback to give user the the amout of users in database, but I don't think it's `necessary/mandatory` thing here, it's just "nice", and we are exposing ourselfs to future problems. 

Just letting the user know that their desired user is created and they can login should be enough.

Reply:
It will be SUPER rare the developer has millions of entries here 😅

I think this is useful in cases where, developer is trying out Backpack, and in the between of the installation process, he is playing around and created an user ...

---

Anyway, @pxpm is there any possible fix for it?
Can we move it inside a try catch block?

Reply:
I don't think it matter at all in this case if he created users or not in the meanwhile. 

He **wanted to create an admin user**, he got the admin user created and can now login with it. 

I agree that is not everyday that you will be having millions of users, unless in that day you are migrating a legacy app to backpack and already have the users setup 👍 

There is no benefit that I can see here by having the user `count()` sorry 😿

Reply:
I like the `count()`, let's keep it. Like @promatik said, I think it's unlikely for the dev to have millions of users on localhost.

Reply:
Errr .. !! @tabacitu you like the "displayed interface", but please take into consideration the underlaying issue and the benefits it brings. 
When that dev with 400k users, an un-optimized db, tries to use backpack on his project, he will see big memory consumption and loading times, for what should be an "easy sql insert".

It can take 10s, no memory limit hit and we are all good, or it can hit timeout / memory limit and break the install script. 
My question is: for Why leave this door open praying for sunny days when it does not add any benefit for the dev experience? 

When user is created we **already give feedback**: User created with name x, email x, pass x ! Add another one ? 

My last arguments, I rest my case here.

Reply:
I don't think counting 400k users will take 10s. I don't think it will even take 1s. And the likelyhood for the db to be unoptimized is slim - after all, we're expecting the standard Laravel `users` table.

I'm afraid I don't agree here Pedro - let's use `count()` for now, and if it turns out I'm wrong (which I often am!) someone will report it, and we can remove or optimize it. But for the 90% use case, I think this will work just fine.